service/history/queues/scheduler.go

```diff
@@ -236,7 +236,8 @@ func (s *schedulerImpl) Start() {
 			func(oldNamespaces, newNamespaces []*namespace.Namespace) {
 				namespaceFailover := false
 				for idx := range oldNamespaces {
-					if oldNamespaces[idx].FailoverVersion() != newNamespaces[idx].FailoverVersion() {
+					if oldNamespaces[idx] == nil ||
+						oldNamespaces[idx].FailoverVersion() != newNamespaces[idx].FailoverVersion() {
 						namespaceFailover = true
```

Comment:
why check old[idx] for nil but not check new[idx] for nil?

Reply:
I can check both. 

But my understanding is only oldNS can be nil when the function is registered: https://github.com/temporalio/temporal/blob/master/common/namespace/registry.go#L321
And later invocation of the callback won't have nil in oldNS: https://github.com/temporalio/temporal/blob/master/common/namespace/registry.go#L517

Reply:
Synced offline. Will revert those changes and skip Scheduler test for sqlite for now to unblock this PR.